tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -334,6 +373,36 @@ private function initializeData(array $initialData, array $expectedData = null)
         }
     }
 
+    private function prepareBulkWriteArguments($arguments)
+    {
+        $operations = [];
+        $operations['options'] = $arguments['options'];
+        foreach ($arguments['requests'] as $requests) {
+            $innerArray = [];
+            switch ($requests['name']) {
+                case 'deleteMany':
+                case 'deleteOne':
+                    $options = array_diff_key($requests['arguments'], ['filter' => 1]);
+                    $innerArray = [$requests['arguments']['filter'], $options];
+                break;
+                case 'insertOne':
+                    $innerArray = [$requests['arguments']['document']];
+                    break;
+                case 'replaceOne':
+                    $options = array_diff_key($requests['arguments'], ['filter' => 1, 'replacement' => 1]);
+                    $innerArray = [$requests['arguments']['filter'], $requests['arguments']['replacement'], $options];
+                break;
+                case 'updateMany':
+                case 'updateOne':
+                    $options = array_diff_key($requests['arguments'], ['filter' => 1, 'update' => 1]);
+                    $innerArray = [$requests['arguments']['filter'], $requests['arguments']['update'], $options];
+                break;
```

Comment:
Can we add a default case to raise an error if we see an unexpected request type?

```
default:
    throw new LogicException('Unsupported bulk write request: ' . $request['name']);
```